### PR TITLE
Deprecate `createMapValue` functions with `Type` parameter

### PR DESCRIPTION
**Pull request by Nadeeshan96**
This PR deprecates the use of `createMapValue` functions with `Type` parameter in the ValueCreator API. Instead of them `createMapValue` functions with `MapType` parameter should be used. This is because in order to create a map value what should be passed is the Map type, not the constrained type of the Map.

Fixes #32052

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java [hunk 1]
```diff
@@ -680,22 +681,47 @@ public static BXml createXmlProcessingInstruction(BString target, BString data,
...
+     * Create a runtime map value with given initial values and given type.
+     *
+     * @param mapType   map type.
+     * @param keyValues initial map values to be populated.
+     * @return map value
+     */
+    public static BMap<BString, Object> createMapValue(MapType mapType, BMapInitialValueEntry[] keyValues) {
+        return new MapValueImpl<>(mapType, keyValues);
+    }
+
```
Comment (gabilang) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java:724]: Shall we add some test cases with this API changes?

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Review by warunalakshitha — CHANGES_REQUESTED** [review]
LGTM

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java [hunk 2]
```diff
@@ -680,22 +681,47 @@ public static BXml createXmlProcessingInstruction(BString target, BString data,
      *
      * @param mapType map type.
      * @return map value
+     * @deprecated use {@link #createMapValue(MapType)} instead
      */
+    @Deprecated
     public static BMap<BString, Object> createMapValue(Type mapType) {
```
Comment (HindujaB) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java:687]: There are some places we use this API to create record values without passing the package information (Eg. : cli args). In those cases, deprecating this API will be wrong. Do we need another API as `createRecordValue(RecordType recordType)`?

Comment (Nadeeshan96) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java:688]: I think so. Hence I added two new methods of `createRecordValue(RecordType recordType ` kind.

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java (hunk 1 above)
Comment (Nadeeshan96) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java:725]: Added tests for newly introduced 4 methods. Will add for other uncovered api methods also.

**Comment by Nadeeshan96** [comment]
> Shall we add a test for new apis?

Added tests for the new API methods and changed the usages of deprecated methods to the new methods.

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/nativeimpl/jvm/runtime/api/tests/Values.java [hunk 3]
```diff
@@ -148,4 +153,43 @@ public static BArray getTypeIds(BObject bObject) {
         }
         return arrayValue;
     }
+
+    public static Object getMapValue1() {
```
Comment (HindujaB) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/nativeimpl/jvm/runtime/api/tests/Values.java:157]: Shall we name these methods as `getMapValue` and `getMapValueWithInitialValues` and similarly for the records?

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java [hunk 4]
```diff
@@ -680,22 +682,47 @@ public static BXml createXmlProcessingInstruction(BString target, BString data,
      *
      * @param mapType map type.
      * @return map value
+     * @deprecated use {@link #createMapValue(MapType)} instead
      */
+    @Deprecated
```
Comment (HindujaB) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java:687]: Shall we create an issue in th https://github.com/ballerina-platform/ballerina-standard-library repo mentioning the API deprecation? Better if we can mention the modules that need the change. 
Similar to https://github.com/ballerina-platform/ballerina-standard-library/issues/1646

Comment (Nadeeshan96) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java:687]: Created https://github.com/ballerina-platform/ballerina-standard-library/issues/2568.

**Review by HindujaB — APPROVED** [review]
LGTM

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**